src/python/gem5/components/boards/riscv_board.py

```diff
@@ -502,14 +502,24 @@ def get_disk_device(self):
 
     @overrides(AbstractSystemBoard)
     def _pre_instantiate(self, full_system: Optional[bool] = None):
-        if len(self._bootloader) > 0:
-            self.workload.bootloader_addr = 0x0
-            self.workload.bootloader_filename = self._bootloader[0]
-            self.workload.kernel_addr = 0x80200000
-            self.workload.entry_point = 0x80000000  # Bootloader starting point
-        else:
-            self.workload.kernel_addr = 0x0
-            self.workload.entry_point = 0x80000000
+        # This is a bit of a hack necessary to get the RiscDemoBoard working
+        # At the time of writing the RiscvBoard does not support SE mode so
+        # this branch looks pointless. However, the RiscvDemoBoard does and
+        # needs this logic in place.
+        #
+        # This should be refactored in the future as part of a chance to have
+        # all boards support both FS and SE modes.
```

Comment:
This is OK, but couldn't you override this in `RiscvDemoBoard` and have something like:

```python
def _pre_instantiate(self...):
     if self._is_fs:
        super()._pre_instantiate()
    else:
        self.workload.kernel_addr = 0x0
...
```
```

Reply:
The problem i have with that is you'd need to copy and paste the `AbstractBoard`'s `_pre_instantiate` function into the `else` block which is a bit annoying. In short, for FS mode we want to run the `RiscvBoard`'s `_pre_instantiate` and for SE mide we just want to run the `AbstractBoard`'s. I tried `super().super()._pre_instantiate()` but Python didn't like it. Perhaps there's a way to do it but in the end i went with the solution I had working.